packages/mobile/src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,349 @@
+  }
+
+  const onBottomSheetClose = () => {
+    setBottomSheetVisible(false)
+  }
+
+  const onHelpPress = () => {
+    setHelpDialogVisible(true)
+  }
+
+  const onNavigationButtonPress = () => {
+    if (!dappSelected) {
+      // Should never happen
+      Logger.warn(TAG, 'Internal error. There was no dapp selected')
+      return
+    }
+    navigateToURI(dappSelected.dappUrl)
+    setBottomSheetVisible(false)
+  }
+
+  return (
+    <SafeAreaView style={styles.safeAreaContainer}>
+      <CustomHeader
+        left={<BackButton />}
+        title={t('dappsScreen.title')}
+        right={
+          <TopBarIconButton
+            icon={<Help />}
+            onPress={onHelpPress}
+            style={styles.helpIconContainer}
+          />
+        }
+      />
+      <BottomSheet isVisible={isBottomSheetVisible} onBackgroundPress={onBottomSheetClose}>
+        <View>
+          <View style={styles.bottomSheetHeader}>
+            <TopBarIconButton icon={<QuitIcon />} onPress={onBottomSheetClose} />
+          </View>
+          <View style={styles.centerContainer}>
+            <Text style={{ ...fontStyles.h2, textAlign: 'center', paddingVertical: 16 }}>
+              {t('dappsScreenBottomSheet.title', { dappName: dappSelected?.name })}
+            </Text>
+            <Text style={{ ...fontStyles.regular, textAlign: 'center', flex: 1 }}>
+              {t('dappsScreenBottomSheet.message')}
+            </Text>
+
+            {/* <Text>Dont show again Message</Text> */}
+            <Button
+              style={styles.bottomSheetButton}
+              onPress={onNavigationButtonPress}
+              text={t('dappsScreenBottomSheet.button', { dappName: dappSelected?.name })}
+            />
+          </View>
+        </View>
+      </BottomSheet>
+
+      <Dialog
+        title={t('dappsScreenHelpDialog.title')}
+        isVisible={isHelpDialogVisible}
+        secondaryActionText={t('dappsScreenHelpDialog.dismiss')}
+        secondaryActionPress={() => setHelpDialogVisible(false)}
+      >
+        {t('dappsScreenHelpDialog.message')}
+      </Dialog>
+
+      <ScrollView style={styles.scrollContainer}>
+        <View style={styles.descriptionContainer}>
+          <Text style={{ ...fontStyles.h1, flex: 1 }}>{t('dappsScreen.message')}</Text>
+          <View style={styles.descriptionImage}>
+            <DappsExplorerLogo />
+          </View>
+        </View>
+        <>
+          {loading && (
+            <View style={styles.centerContainer}>
+              <ActivityIndicator
+                style={styles.loadingIcon}
+                size="large"
+                color={colors.greenBrand}
+                testID="DAppExplorerScreen/loading"
+              />
+            </View>
+          )}
+          {!loading &&
+            categoriesWithItems.map((category) => renderCategoryWithItems(category, onItemPress))}
+        </>
+      </ScrollView>
+    </SafeAreaView>
+  )
+}
+
+function renderCategoryWithItems(
```

Comment:
was wondering if we have any preferences for patterns here...this way of course is fine and works :) but it is equivalent to doing `function CategoryWithItems(props) {...}` and calling it in line 198 like 
```
categoriesWithItems.map(
  <CategoryWithItems
    {...someProps}
  />
)}
```
this function is producing a react component so i personally prefer invoking it like a component but not sure what the consensus is, so thought i'd ask! we do use the `renderXXX` function pattern if the `renderXXX` is inside a component/class already which makes sense, but this one is outside and standalone.

Reply:
Yes! When I was doing it I thought that using react component would be better, I'll change it :)